src/perception/sythesis/cone_mapping/launch/cone_mapping.launch.py

```diff
@@ -0,0 +1,44 @@
+from launch import LaunchDescription
+from launch_ros.actions import Node
+from launch.actions.declare_launch_argument import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
+from ament_index_python import get_package_prefix
+from launch.actions import OpaqueFunction
+import os
+
+def generate_launch_description():
+    package_name = 'cone_mapping'
+    package_dir = os.path.join(get_package_prefix(package_name), 'lib', package_name)   # directory of installed node names
+    mappers = os.listdir(package_dir)  # retrieves installed node names 
+
+    launch_descriptions = [        
+        # launch arguments
+        DeclareLaunchArgument(
+            'node_name',
+            default_value='kalman_filter',
+            description='which node from cone_mapping package to launch'
+        )
+    ]
+
+    node = OpaqueFunction(function=get_node, args=[package_name, mappers])    # get a list of actions

```

Comment:
Didn't know this was a thing. Looks really clean, I'll have to start using it myself.